lxa_iobus/object_directory.py

```diff
@@ -0,0 +1,608 @@
+import logging
+import struct
+import types
+
+from lxa_iobus.canopen import SdoAbort
+
+logger = logging.getLogger("lxa_iobus.object_directory")
+
+
+class ProtocolVersionError(Exception):
+    pass
+
+
+class SubIndex(object):
+    @classmethod
```

Comment:
Sounds good!

I've been stressing myself to get this PR out, because progress went slower than I would have liked and that's why I have skipped adding comments (and tests).

The `object_directory.py` would be used by anyone wanting to use `lxa_iobus` as a library, so it should really be documented a bit better. Will do.

Reply:
I've gone through the file and documented / commented wherever I saw fit.
Please review again.